### PR TITLE
Fix Issue #320 Fix transparent object ordering in 3d

### PR DESCRIPTION
**Pull request by Katsutoshii**
Solves https://github.com/djeedai/bevy_hanabi/issues/320

### CODE REVIEW

**Comment by Katsutoshii** [comment]
Tested with `cargo run --example ordering --features="bevy/bevy_winit bevy/bevy_pbr bevy/png 3d examples_world_inspector"`

**Review by djeedai — COMMENTED** [review]
Thanks for the fix! Just a few minor things.

File: src/render/batch.rs [hunk 1]
```diff
@@ -81,6 +81,10 @@ pub(crate) struct EffectDrawBatch {
     /// rendering.
     #[cfg(feature = "2d")]
     pub z_sort_key_2d: FloatOrd,
+    /// For 3d rendering, the distance from the camera used for ordering.
```
[the remark this answers is not in the document]
Comment (Katsutoshii) [reply src/render/batch.rs:84]: True, updated the comment.

File: src/render/mod.rs [hunk 2]
```diff
@@ -2154,6 +2166,13 @@ pub(crate) fn prepare_effects(
         #[cfg(feature = "2d")]
         let z_sort_key_2d = input.z_sort_key_2d;
 
+        #[cfg(feature = "3d")]
+        let translation_3d = {
+            let (_scale, _rotation, translation) =
+                Mat4::from(&input.transform).to_scale_rotation_translation();
```
[the remark this answers is not in the document]
Comment (Katsutoshii) [reply src/render/mod.rs:2172]: Updated to avoid the decomposition.

**Comment by djeedai** [comment]
I've changed the example a bit to make sure we can see the blue rectangle being rendered on top and partially (but not completely) affecting the particles:

image

**Review by djeedai — APPROVED** [review]

**Comment by Katsutoshii** [comment]
Great, thanks!

**Pull request merged**